From: Mark McLoughlin <markmc@redhat.com>
Date: Mon, 27 Apr 2009 10:18:14 +0100
Subject: [PATCH] kvm: qemu: framebuffer: build fix for target-arm

Include qemu-kvm.h for non-KVM_UPSTREAM building and surround the
kvm code with USE_KVM guards.

Fixes target-arm:

 hw/framebuffer.c: In function 'framebuffer_update_display':
 hw/framebuffer.c:53: warning: implicit declaration of function 'kvm_enabled'
 hw/framebuffer.c:54: warning: implicit declaration of function 'kvm_physical_sync_dirty_bitmap'

Signed-off-by: Mark McLoughlin <markmc@redhat.com>
---
 hw/framebuffer.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/hw/framebuffer.c b/hw/framebuffer.c
index 1086ba9..e2d7604 100644
--- a/hw/framebuffer.c
+++ b/hw/framebuffer.c
@@ -18,6 +18,7 @@
 #include "console.h"
 #include "framebuffer.h"
 #include "kvm.h"
+#include "qemu-kvm.h"
 
 /* Render an image from a shared memory framebuffer.  */
    
@@ -50,9 +51,11 @@ void framebuffer_update_display(
     *first_row = -1;
     src_len = src_width * rows;
 
+#ifdef USE_KVM
     if (kvm_enabled()) {
         kvm_physical_sync_dirty_bitmap(base, src_len);
     }
+#endif
     pd = cpu_get_physical_page_desc(base);
     pd2 = cpu_get_physical_page_desc(base + src_len - 1);
     /* We should reall check that this is a continuous ram region.
-- 
1.6.0.6

